core/src/main/kotlin/io/specmatic/core/HttpRequestPattern.kt

```diff
@@ -30,6 +30,10 @@ data class HttpRequestPattern(
     val multiPartFormDataPattern: List<MultiPartFormDataPattern> = emptyList(),
     val securitySchemes: List<OpenAPISecurityScheme> = listOf(NoSecurityScheme())
 ) {
+    fun getHeaderKeys() = headersPattern.pattern.keys
+
+    fun getQueryParamKeys() = httpQueryParamPattern.queryPatterns.keys
```

Comment:
Similar to the previous comment, httpQueryParamPattern.queryParamNames

Reply:
Addressed.